src/components/application_manager/src/commands/command_impl.cc

```diff
@@ -94,14 +94,59 @@ void CommandImpl::SetAllowedToTerminate(const bool allowed) {
   allowed_to_terminate_ = allowed;
 }
 
+bool CommandImpl::ReplaceMobileWithHMIAppId(
+    NsSmartDeviceLink::NsSmartObjects::SmartObject& message) {
+  if (message.keyExists(strings::app_id)) {
+    ApplicationSharedPtr application =
+        application_manager_.application(message[strings::app_id].asUInt());
+    if (!application) {
+      LOG4CXX_ERROR(logger_, "Substitution mobile --> HMI id is failed.");
+      return false;
+    }
+    LOG4CXX_DEBUG(logger_,
+                  "ReplaceMobileWithHMIAppId from "
+                      << message[strings::app_id].asInt() << " to "
+                      << application->hmi_app_id());
+    message[strings::app_id] = application->hmi_app_id();
+  } else {
+    switch (message.getType()) {
+      case smart_objects::SmartType::SmartType_Array: {
+        smart_objects::SmartArray* message_array = message.asArray();
+        smart_objects::SmartArray::iterator it = message_array->begin();
+        for (; it != message_array->end(); ++it) {
+          if (!ReplaceMobileWithHMIAppId(*it)) {
+            break;
+          }
+        }
+        break;
+      }
+      case smart_objects::SmartType::SmartType_Map: {
+        std::set<std::string> keys = message.enumerate();
+        std::set<std::string>::const_iterator key = keys.begin();
+        for (; key != keys.end(); ++key) {
+          std::string k = *key;
+          if (!ReplaceMobileWithHMIAppId(message[*key])) {
+            break;
+          }
+        }
+        break;
+      }
+      default: { break; }
+    }
+  }
+
+  return true;
+}
+
+// DEPRECATED
 void CommandImpl::ReplaceMobileByHMIAppId(
```

Comment:
You could change this to just use the new version of the function:

```
void CommandImpl::ReplaceMobileByHMIAppId(
    NsSmartDeviceLink::NsSmartObjects::SmartObject& message) {
  if (ReplaceMobileWithHMIAppId(message)) {
    //Log Error
  }
}
```

Better than having the function duplicated.

Reply:
@jacobkeeler Anyway we need to use new function which returns `bool` . We use it in `bool NotificationToHMI::Init()`. Therefore what is the sense to make wrapper if we can not avoid direct usage of bool version of this function?

Reply:
I'm not saying that this version of the function shouldn't be deprecated. This change would be used primarily to simplify the diff, as well as reduce code bloat. It would be much cleaner to read if it was refactored to use it's replacement, rather than just leaving a outdated copy of the function. 

Considering minimal effort is needed to create this, I think it would be a worthwhile improvement.

Reply:
@jacobkeeler Let's summarize our discussion. Please confirm thast my understanding of your idea is correct or wrong. 
- We will have separate non-deprecated method `bool ReplaceMobileWithHMIAppId`
- We will have the `DEPRECATED` wrapper `void CommandImpl::ReplaceMobileByHMIAppId` which will use `bool ReplaceMobileWithHMIAppId` inside its body. 
- When it's time to remove all deprecated methods - we will have to remove all pure deprecated methods as well as deprecated wrappers.
- Due to previous point - we have to change usage of all deprecated wrappers to the separate non-deprecated functions which were used inside the wrappers.

Are these points correct? 

Thank you for reply.

Reply:
@jacobkeeler Fixed in 88fc4e0d5a15d60bd2edfc9bd623a54bd71d0243

Reply:
Regarding point 4:
> we have to change usage of all deprecated wrappers to the separate non-deprecated functions which were used inside the wrappers

You should not be using these wrappers at all, that is why they are deprecated. This refactoring is primarily for readability.

As an additional point, one benefit to this approach is that, if an OEM is using the outdated version of this function in their own software, any fixes to the new version of the function carry over to the deprecated function (as would be the case if the function signature was unchanged, but the internals were modified).

Reply:
@jacobkeeler Fixed in b969449ecb89eb95e288d9d0024d1b1d2d1f66f0